diff -up xf86-video-vmware-13.0.1/src/vmwarevideo.c.jx xf86-video-vmware-13.0.1/src/vmwarevideo.c
--- xf86-video-vmware-13.0.1/src/vmwarevideo.c.jx	2012-08-06 18:31:14.000000000 -0400
+++ xf86-video-vmware-13.0.1/src/vmwarevideo.c	2014-08-14 09:51:15.960270366 -0400
@@ -789,7 +789,8 @@ vmwareVideoPlay(ScrnInfoPtr pScrn, VMWAR
     struct PACKED _body {
         uint32 escape;
         uint32 streamId;
-        struct _item items[SVGA_VIDEO_NUM_REGS];
+        /* Old hosts can not handle more then these regs */
+        struct _item items[SVGA_VIDEO_DATA_GMRID];
     };
 
     struct PACKED _cmdSetRegs {
@@ -831,7 +832,7 @@ vmwareVideoPlay(ScrnInfoPtr pScrn, VMWAR
     cmdSetRegs.body.streamId = pVid->streamId;
 
     items = cmdSetRegs.body.items;
-    for (i = SVGA_VIDEO_ENABLED; i < SVGA_VIDEO_NUM_REGS; i++) {
+    for (i = SVGA_VIDEO_ENABLED; i < SVGA_VIDEO_DATA_GMRID; i++) {
         items[i].regId = i;
     }
 
